src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (num) => {
+    if (num !== undefined) {
+      sum += num;
+
+      return adder;
+    } else {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
```

Comment:
```suggestion
    if (num !== undefined) {
      sum += num;

      return adder;
    } 
    
    const result = sum;

    sum = 0;

    return result;
```
`else` block is redundant when you return inside `if` block - FYI